cli/bblfshctl/cmd/base.go

```diff
@@ -7,27 +7,46 @@ import (
 	"github.com/bblfsh/server/daemon/protocol"
 
 	"google.golang.org/grpc"
+	sdk "gopkg.in/bblfsh/sdk.v1/protocol"
 )
 
-type GRPCCommand struct {
-	Network string `long:"endpoint" default:"unix" description:"control server network type"`
-	Address string `long:"address" default:"/var/run/bblfshctl.sock" description:"control server address to connect"`
+type ControlCommand struct {
+	Network string `long:"ctl-endpoint" default:"unix" description:"control server network type"`
+	Address string `long:"ctl-address" default:"/var/run/bblfshctl.sock" description:"control server address to connect"`
 
 	conn *grpc.ClientConn
 	srv  protocol.ProtocolServiceClient
 }
 
-func (c *GRPCCommand) Execute(args []string) error {
-	conn, err := grpc.Dial(c.Address,
+func (c *ControlCommand) Execute(args []string) error {
+	conn, err := dialGRPC(c.Network, c.Address)
+	c.conn = conn
+	c.srv = protocol.NewProtocolServiceClient(conn)
+	return err
+}
+
+type UserCommand struct {
+	Network string `long:"endpoint" default:"tcp" description:"server network type"`
+	Address string `long:"address" default:"localhost:9432" description:"server address to connect"`
+
+	conn *grpc.ClientConn
+	srv  sdk.ProtocolServiceClient
+}
+
+func (c *UserCommand) Execute(args []string) error {
+	conn, err := dialGRPC(c.Network, c.Address)
```

Comment:
Slightly simpler:

```go
c.conn, err = ...
c.srv = sdk....(c.conn)
```

Reply:
the same number of line :p

Reply:
It should be a line less, but yes, it's a _slight_ improvement =)